.github/workflows/ci.yml

```diff
@@ -0,0 +1,319 @@
+name: Continuous Integration
+env:
+  DOCKER_IMAGE: usabillabv/php
+  DOCKER_BUILDKIT: 1
+  DOCKER_CLI_EXPERIMENTAL: enabled
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+jobs:
+  generate-alpine-strategy:
+    name: Generate Alpine
+    runs-on: ubuntu-latest
+    outputs:
+      alpine: ${{ steps.generate-alpine-strategy.outputs.versions }}
+    steps:
+      - id: generate-alpine-strategy
+        name: Generate Alpine
+        run: |
+          echo "::set-output name=versions::[\"3.12\", \"3.11\", \"3.10\"]"
+  generate-nginx-strategy:
+    name: Generate nginx
+    runs-on: ubuntu-latest
+    outputs:
+      nginx: ${{ steps.generate-nginx-strategy.outputs.versions }}
+    steps:
+      - id: generate-nginx-strategy
+        name: Generate nginx
+        run: |
+          echo "::set-output name=versions::[\"1.17-nginx1-nginx\", \"1.16\", \"1.15\", \"1.14\"]"
```

Comment:
That is because artefact names can't container spaces. We're pulling it apart here: https://github.com/usabilla/php-docker-template/pull/160/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR109

Reply:
Hmmm ok, maybe worth putting a comment for the "hack", then we can find a better solution

Reply:
This hack is more a feature hiding in plain sight than a hack. In a future follow up I want to consolidate all build/test/scan/push steps into one, and by using a string with delimiter we can easily compile a matrix that puts what to build and test in a formatted string. Done the same thing for my own PHP images and it keeps the workflow file a lot smaller. But since doing so is an endeavour out of scope for this migration PR we decided to go with the current approach before looking into optimizations like that. I'll add that comment :+1: .